src/main/java/de/zalando/aruha/nakadi/repository/kafka/KafkaTopicRepository.java

```diff
@@ -197,9 +197,17 @@ public void syncPostBatch(final String topicId, final List<BatchItem> batch) thr
             final boolean isSuccessful = done.await(settings.getKafkaSendTimeoutMs(), TimeUnit.MILLISECONDS);
 
             if (!isSuccessful) {
+                for (final BatchItem item : batch) {
+                    item.safeFail("timed out");
```

Comment:
ah ok. Now I see that safeFail will only set new status if event was not submitted.
Everything should be fine then.